eng/Directory.Build.Common.targets

```diff
@@ -39,7 +39,7 @@
   </Target>
 
   <!--Run Aggregate Updates to Source -->
-  <Target Name="UpdateSourceOnBuild" AfterTargets="Build" Condition="'$(UpdateSourceOnBuild)' == 'true' and '$(IsShippingClientLibrary)' == 'true'" >
+  <Target Name="UpdateSourceOnBuild" AfterTargets="Build" Condition="'$(UpdateSourceOnBuild)' == 'true' and '$(IsShippingClientLibrary)' == 'true' and $(MSBuildProjectDirectory.Contains($(ServiceDirectory)))" >
```

Comment:
How does this help fix the race conditions?

Reply:
It is supposed to make it so that it only runs for the service specified. i.e if KeyVault or Storage is ran and it pull in Core the build does not run this target for core. However this will not be required once SnippetGenerator can be consumed from the feed.